src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -193,6 +193,13 @@ private void InitializeContext(K8SConfiguration k8SConfig, string currentContext
                 throw new KubeConfigException($"CurrentContext: {currentContext} not found in contexts in kubeconfig");
             }
 
+            if (string.IsNullOrEmpty(activeContext.ContextDetails?.Cluster))
+            {
+                // This serves as validation for any of the properties of ContextDetails being set.
+                // Other locations in code assume that ContextDetails is non-null.
+                throw new KubeConfigException($"Cluster not set for context `{currentContext}` in kubeconfig");
+            }
```

Comment:
Added this validation because I see a bunch of code running after this at assumes `activeContext.ContextDetails` is null, and it will crash with a null-ref based on invalid config data.